File: backends/Contracts/Common.Infra/Cache/Redis/Handlers/Set.cs
using D2.Contracts.Common;
using D2.Contracts.Common.App;
using S = D2.Contracts.Common.App.ICacheService;

namespace Common.Infra.Cache.Redis.Handlers;

/// <inheritdoc cref="S.ISetHandler{TValue}"/>
public class Set<TValue> : BaseHandler<
        S.ISetHandler<TValue>, S.SetInput<TValue>, S.SetOutput>,
    S.ISetHandler<TValue>
{
    /// <summary>
    /// Initializes a new instance of the <see cref="Set{TValue}"/> class.
    /// </summary>
    ///
    /// <inheritdoc/>
    public Set(IHandlerContext context) : base(context) { }

    /// <inheritdoc/>
    protected override async ValueTask<D2Result<S.SetOutput?>> ExecuteAsync(
        S.SetInput<TValue> input,
        CancellationToken ct = default)
    {
        // TODO: Implement.
        return default;
    }
}

File: backends/node/shared/interfaces/src/caching/distributed/handlers/u/set.ts
import type { IHandler } from "@d2/handler";

/** Input for setting a single value in the distributed cache. */
export interface SetInput<TValue> {
  key: string;
  value: TValue;
  expirationMs?: number;
}

/** Output for setting a single value in the distributed cache. */

export interface SetOutput {}

/** Handler for setting a single value in the distributed cache. */
export type ISetHandler<TValue> = IHandler<SetInput<TValue>, SetOutput>;
